### DISCUSSION TITLE
Task definition file not found for deploying to AWS ECS

### DISCUSSION THREAD

**Discussion by naveenmarthala** [community/community#26431] [Actions]
<p>I am trying to push docker images from GitHub Actions to AWS ECR then to ECS with a task to deploy the latest container in AWS ECS.</p>
<p>I am using the default aws.config file that AWS provides on GitHub Marketplace to push docker images to AWS ECR, which could then to deployed as containers to AWS ECS by defining tasks. And I got an error message with defining tasks to deploy to ECS(in the section “Fill in the new image ID in the Amazon ECS task definition” in aws.yml) and it is:</p>
<pre><code class="lang-auto">Error: Task definition file does not exist: task-def.json
</code></pre>
<p>Some information to prove everything is in place(presumably):<br>
<strong>Here’s me checking the existence of file and pushing the task from my terminal to AWS ECS:</strong><br>
<strong>Here’s me checking the existence of file and pushing the task from my terminal to AWS ECS:</strong><br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/4/6/46de923dcf1952ba60db4b3d4ec1a217a60fc458.png" data-download-href="/uploads/short-url/a6WhkXNFRilhNIqHO5nExpDfiRy.png?dl=1" title="Screenshot from 2020-10-26 13-11-55" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/41302653/181097629-1b308c0e-7c92-410d-bb16-bce6969b3f5b.png" alt="Screenshot from 2020-10-26 13-11-55" data-base62-sha1="a6WhkXNFRilhNIqHO5nExpDfiRy" width="690" height="268" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/4/6/46de923dcf1952ba60db4b3d4ec1a217a60fc458_2_690x268.png, https://aws1.discourse-cdn.com/github/optimized/3X/4/6/46de923dcf1952ba60db4b3d4ec1a217a60fc458_2_1035x402.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/4/6/46de923dcf1952ba60db4b3d4ec1a217a60fc458_2_1380x536.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/4/6/46de923dcf1952ba60db4b3d4ec1a217a60fc458_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot from 2020-10-26 13-11-55</span><span class="informations">1803×702 73.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>And here’s the part of my “.github/workflows/aws.yml” in my repo for task definition (full file can be found <a href="https://github.com/naveen-9697/docker-ci-cd-aws-practice/blob/cca33393893155a0832faf59dec2d762ff75f25b/.github/workflows/aws.yml" rel="noopener nofollow ugc">here in my repository</a>):</p>
<pre><code class="lang-auto">    - name: Fill in the new image ID in the Amazon ECS task definition
      id: task-def
      uses: aws-actions/amazon-ecs-render-task-definition@v1
      with:
        task-definition: ecs-task-definition.json
        container-name: aws-dock-p-cont
        image: ${{ steps.build-image.outputs.image }}
</code></pre>
<p>Images are being pushed to my ECR repository without any problems, but the problem comes during executing tasks to deploy and start containers.</p>
<p><strong>So, as you can see task-definition file does exist, but GitHub doesn’t recongise it. What has caused this error and how do I fix this?</strong></p>
<p>I couldn’t post more screenshots. So, for all the information: <a href="https://github.com/aws-actions/amazon-ecs-render-task-definition/issues/68#issue-729342570" rel="noopener nofollow ugc">https://github.com/aws-actions/amazon-ecs-render-task-definition/issues/68#issue-729342570</a></p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/naveen-9697">@naveen-9697</a>,</p>
<p>You can try to check with the steps below:</p>
<ol>
<li>
<p>Before the “<strong>Fill in the new image ID in the Amazon ECS task definition</strong>” step, set up a step to like as this.</p>
<pre><code class="lang-yaml">- name: List files
  run: ls -R
</code></pre>
<p>This will list all the files in the working directory.</p>
</li>
<li>
<p>From the listed files, check if the task definition file (<strong>ecs-task-definition.json</strong>) is existing.</p>
<ul>
<li>
<p>If task definition file is existing, the problem should be that the “<strong>aws-actions/amazon-ecs-render-task-definition</strong>” action can’t recognize this file. There may be some issues occur the action. If so, I recommend that you can report an issue the appropriate engineering team for further investigation and evaluation on the <a href="https://github.com/aws-actions/amazon-ecs-render-task-definition/issues" rel="noopener nofollow ugc">action repository</a>.</p>
</li>
<li>
<p>If the definition file isn’t existing, make sure it indeed is existing in the source files of your repository and have checkout the source files.<br>
If it is indeed existing in the repository, maybe the previous steps have deleted the task definition file.</p>
</li>
</ul>
</li>
</ol>

**Comment by naveenmarthala** [comment, marked as answer]
<p>Thank you very much for trying to help me.</p>
<p>I fixed it. It had happened because, there was no file named ‘ecs-task-definition.json’ in github repo. I haven’t uploaded it with the thought that github will pull it from AWS.</p>
